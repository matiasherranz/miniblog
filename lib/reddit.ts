import axios from 'axios';
import { createAsyncThunk } from '@reduxjs/toolkit';

import { RedditStateType } from '../util/types';
import { BASE_REDDIT_URL, PAGE_SIZE } from '../util/constants';

type ParamsType = Record<string, string>;

export const fetchRedditPosts = createAsyncThunk<
  Promise<string>,
  void,
  { state: RedditStateType }
>(
  'notes/fetchPosts',
  async (_, { getState, rejectWithValue }) => {
    console.log('FETCHING POSTS');
    try {
      const { reddit } = getState();
      const params: ParamsType = {
        limit: PAGE_SIZE.toString(),
      };

      if (reddit.after) {
        params.after = reddit.after;
      }

      if (!reddit.hasMore) {
        throw new Error('No more posts available');
      }

      const response = await axios.get(`${BASE_REDDIT_URL}/top.json`, {
        params,
      });
      return response.data;
    } catch (error) {
      return rejectWithValue({ error: error.message });
    }
  },
  {
    condition: (_, { getState }) => {
      const { reddit } = getState();

      // Already fetching, cancel this new request
      return reddit.loading !== 'loading';
    },
  }
);
