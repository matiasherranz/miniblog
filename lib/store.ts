import { configureStore } from '@reduxjs/toolkit';
import { persistStore } from 'redux-persist';
import ReduxThunk from 'redux-thunk';

import redditReducer from './slices/redditSlice';

export const store = configureStore({
  reducer: {
    reddit: redditReducer,
  },
  devTools: true,
  middleware: [ReduxThunk],
});

export const persistor = persistStore(store);
