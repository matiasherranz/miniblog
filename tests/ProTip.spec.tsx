import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import { store } from '../lib/store';
import ProTip from '../components/ProTip';

describe('ProTip', () => {
  const getWrapper = () =>
    mount(
      <Provider store={store}>
        <ProTip />
      </Provider>
    );

  test('ProTip component renders properly', () => {
    const wrapper = getWrapper();

    expect(wrapper).toMatchSnapshot();
  });
});
