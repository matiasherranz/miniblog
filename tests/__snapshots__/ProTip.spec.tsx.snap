// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProTip ProTip component renders properly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ProTip>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-root-1"
      color="textSecondary"
    >
      <ForwardRef(Typography)
        className="makeStyles-root-1"
        classes={
          Object {
            "alignCenter": "MuiTypography-alignCenter",
            "alignJustify": "MuiTypography-alignJustify",
            "alignLeft": "MuiTypography-alignLeft",
            "alignRight": "MuiTypography-alignRight",
            "body1": "MuiTypography-body1",
            "body2": "MuiTypography-body2",
            "button": "MuiTypography-button",
            "caption": "MuiTypography-caption",
            "colorError": "MuiTypography-colorError",
            "colorInherit": "MuiTypography-colorInherit",
            "colorPrimary": "MuiTypography-colorPrimary",
            "colorSecondary": "MuiTypography-colorSecondary",
            "colorTextPrimary": "MuiTypography-colorTextPrimary",
            "colorTextSecondary": "MuiTypography-colorTextSecondary",
            "displayBlock": "MuiTypography-displayBlock",
            "displayInline": "MuiTypography-displayInline",
            "gutterBottom": "MuiTypography-gutterBottom",
            "h1": "MuiTypography-h1",
            "h2": "MuiTypography-h2",
            "h3": "MuiTypography-h3",
            "h4": "MuiTypography-h4",
            "h5": "MuiTypography-h5",
            "h6": "MuiTypography-h6",
            "noWrap": "MuiTypography-noWrap",
            "overline": "MuiTypography-overline",
            "paragraph": "MuiTypography-paragraph",
            "root": "MuiTypography-root",
            "srOnly": "MuiTypography-srOnly",
            "subtitle1": "MuiTypography-subtitle1",
            "subtitle2": "MuiTypography-subtitle2",
          }
        }
        color="textSecondary"
      >
        <p
          className="MuiTypography-root makeStyles-root-1 MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          <LightBulbIcon
            className="makeStyles-lightBulb-2"
          >
            <WithStyles(ForwardRef(SvgIcon))
              className="makeStyles-lightBulb-2"
            >
              <ForwardRef(SvgIcon)
                className="makeStyles-lightBulb-2"
                classes={
                  Object {
                    "colorAction": "MuiSvgIcon-colorAction",
                    "colorDisabled": "MuiSvgIcon-colorDisabled",
                    "colorError": "MuiSvgIcon-colorError",
                    "colorPrimary": "MuiSvgIcon-colorPrimary",
                    "colorSecondary": "MuiSvgIcon-colorSecondary",
                    "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                    "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                    "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                    "root": "MuiSvgIcon-root",
                  }
                }
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root makeStyles-lightBulb-2"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 21c0 .55.45 1 1 1h4c.55 0 1-.45 1-1v-1H9v1zm3-19C8.14 2 5 5.14 5 9c0 2.38 1.19 4.47 3 5.74V17c0 .55.45 1 1 1h6c.55 0 1-.45 1-1v-2.26c1.81-1.27 3-3.36 3-5.74 0-3.86-3.14-7-7-7zm2.85 11.1l-.85.6V16h-4v-2.3l-.85-.6C7.8 12.16 7 10.63 7 9c0-2.76 2.24-5 5-5s5 2.24 5 5c0 1.63-.8 3.16-2.15 4.1z"
                  />
                </svg>
              </ForwardRef(SvgIcon)>
            </WithStyles(ForwardRef(SvgIcon))>
          </LightBulbIcon>
          Pro tip: Want to see another NextJS site?
           
          <WithStyles(ForwardRef(Link))
            href="https://nextjs-blog.matiasherranz.vercel.app"
          >
            <ForwardRef(Link)
              classes={
                Object {
                  "button": "MuiLink-button",
                  "focusVisible": "Mui-focusVisible",
                  "root": "MuiLink-root",
                  "underlineAlways": "MuiLink-underlineAlways",
                  "underlineHover": "MuiLink-underlineHover",
                  "underlineNone": "MuiLink-underlineNone",
                }
              }
              href="https://nextjs-blog.matiasherranz.vercel.app"
            >
              <WithStyles(ForwardRef(Typography))
                className="MuiLink-root MuiLink-underlineHover"
                color="primary"
                component="a"
                href="https://nextjs-blog.matiasherranz.vercel.app"
                onBlur={[Function]}
                onFocus={[Function]}
                variant="inherit"
              >
                <ForwardRef(Typography)
                  className="MuiLink-root MuiLink-underlineHover"
                  classes={
                    Object {
                      "alignCenter": "MuiTypography-alignCenter",
                      "alignJustify": "MuiTypography-alignJustify",
                      "alignLeft": "MuiTypography-alignLeft",
                      "alignRight": "MuiTypography-alignRight",
                      "body1": "MuiTypography-body1",
                      "body2": "MuiTypography-body2",
                      "button": "MuiTypography-button",
                      "caption": "MuiTypography-caption",
                      "colorError": "MuiTypography-colorError",
                      "colorInherit": "MuiTypography-colorInherit",
                      "colorPrimary": "MuiTypography-colorPrimary",
                      "colorSecondary": "MuiTypography-colorSecondary",
                      "colorTextPrimary": "MuiTypography-colorTextPrimary",
                      "colorTextSecondary": "MuiTypography-colorTextSecondary",
                      "displayBlock": "MuiTypography-displayBlock",
                      "displayInline": "MuiTypography-displayInline",
                      "gutterBottom": "MuiTypography-gutterBottom",
                      "h1": "MuiTypography-h1",
                      "h2": "MuiTypography-h2",
                      "h3": "MuiTypography-h3",
                      "h4": "MuiTypography-h4",
                      "h5": "MuiTypography-h5",
                      "h6": "MuiTypography-h6",
                      "noWrap": "MuiTypography-noWrap",
                      "overline": "MuiTypography-overline",
                      "paragraph": "MuiTypography-paragraph",
                      "root": "MuiTypography-root",
                      "srOnly": "MuiTypography-srOnly",
                      "subtitle1": "MuiTypography-subtitle1",
                      "subtitle2": "MuiTypography-subtitle2",
                    }
                  }
                  color="primary"
                  component="a"
                  href="https://nextjs-blog.matiasherranz.vercel.app"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  variant="inherit"
                >
                  <a
                    className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                    href="https://nextjs-blog.matiasherranz.vercel.app"
                    onBlur={[Function]}
                    onFocus={[Function]}
                  >
                    Here's one!
                  </a>
                </ForwardRef(Typography)>
              </WithStyles(ForwardRef(Typography))>
            </ForwardRef(Link)>
          </WithStyles(ForwardRef(Link))>
        </p>
      </ForwardRef(Typography)>
    </WithStyles(ForwardRef(Typography))>
  </ProTip>
</Provider>
`;
